### PR TITLE
Grab lerping on synchronized clients

### PR DESCRIPTION
**Pull request by tombuMS**
This adds in lerping for the synchronized clients during a grab.  This is performed through the existing actor correction payload that is no fully hooked up and supported.

Referenced by issue #[320](https://github.com/microsoft/mixed-reality-extension-sdk/issues/320)

### CODE REVIEW

File: packages/sdk/src/adapters/multipeer/rules.ts [hunk 1]
```diff
@@ -257,35 +257,75 @@ export const Rules: { [id in Payloads.PayloadType]: Rule } = {
...
+                    existingPayload.actor = deepmerge(existingPayload.actor,  {
+                        payload: {
+                            actor: {
+                                transform: {
+                                    app: message.payload.appTransform
+                                }
+                            }
+                        }
+                    });
+                    return undefined;
```
Comment (eanders-MS) [inline packages/sdk/src/adapters/multipeer/rules.ts]: >return undefined; [](start = 20, length = 17)

Can you document why you're returning undefined here?

Comment (tombuMS) [reply packages/sdk/src/adapters/multipeer/rules.ts]: I got this from the actor-update so I will update with a comment there as well.

---
In reply to: [294896385](https://github.com/microsoft/mixed-reality-extension-sdk/pull/350#discussion_r294896385) [](ancestors = 294896385)

File: packages/sdk/src/adapters/multipeer/rules.ts [hunk 2]
```diff
@@ -257,35 +257,75 @@ export const Rules: { [id in Payloads.PayloadType]: Rule } = {
...
             ) => {
-                message.payload.type = 'actor-update';
-                client.queueMessage(message, promise);
-                return undefined;
+                // Coalesce this actor correction with the previously queued update if it exists, maintaining a single
+                // update for this actor rather than queuing a series of them.  This is fine, as we do not need to lerp
+                // an actor correction on a late join user.  It can just be the updated actor values.
+                const payload = message.payload;
+                const queuedMessage = client.queuedMessages
+                    .filter(value =>
```
Comment (stevenvergenz) [inline packages/sdk/src/adapters/multipeer/rules.ts]: >filter [](start = 21, length = 6)

Why not a simple `find` instead of `filter` and `shift`?

Comment (tombuMS) [reply packages/sdk/src/adapters/multipeer/rules.ts]: I simply replicated what was already wrote for the actor-update rule.  @eanders-MS would be a better person to answer his original reasoning in doing it this way.

Comment (eanders-MS) [reply packages/sdk/src/adapters/multipeer/rules.ts]: An oversight on my part for not commenting it initially. Thanks for correcting it.

**Pull request merged**